dbdimp.c

```diff
@@ -1855,6 +1855,29 @@ static bool mariadb_dr_connect(
           }
         }
 
+        (void)hv_stores(processed, "mariadb_auth_plugin", &PL_sv_yes);
+        if ((svp = hv_fetchs(hv, "mariadb_auth_plugin", FALSE)) && *svp)
+        {
+          int error = 1;
+          STRLEN len;
+          char *auth_plugin = SvPVutf8(*svp, len);
+          if (strlen(auth_plugin) != len)
+          {
+            error_nul_character(dbh, "mariadb_auth_plugin");
+            mariadb_db_disconnect(dbh, imp_dbh);
+            return FALSE;
+          }
+#if MYSQL_VERSION_ID >= 50507 && MYSQL_VERSION_ID != 60000
+          error = mysql_options(sock, MYSQL_DEFAULT_AUTH, auth_plugin);
+#endif
+          if (error)
+          {
+            mariadb_dr_do_error(dbh, CR_CONNECTION_ERROR, "Connection error: mariadb_auth_plugin is not supported", "HY000");
+            mariadb_db_disconnect(dbh, imp_dbh);
+            return FALSE;
+          }
+        }
+
```

Comment:
Again, what about testing?

Reply:
Fully automated testing of `mariadb_auth_plugin` is not possible. It requires configuration of _different_ authentication method which would be understood by both client and server. Maybe it could be possible to write github CI testing for it, but that would require to change / hack how MySQL::Sandbox setup mysql server in its virtual env.

Reply:
Not even the `mysql_native_password` with a new client?

Reply:
We cannot expect that `mysql_native_password` is always present in new mysql versions (e.g. system admin can decide to not install support for it) and also we cannot expect that oracle does not decide to drop support for this specific plugin in some new mysql version. So automated test which is always run during install/test phase of perl DBD-MariaDB and uses `mysql_native_password` is fragile and can cause breaking of installation in future.

We can run such test only in controlled env, e.g. on github actions under configuration where we are sure that this one specific method `mysql_native_password` is enabled and is working.